test/e2e/prometheus_test.go

```diff
@@ -0,0 +1,312 @@
+	"antrea_controller_length_address_group_queue",
+	"antrea_controller_length_applied_to_group_queue",
+	"antrea_controller_length_network_policy_queue",
+	"antrea_controller_network_policy_processed",
+	"antrea_controller_network_policy_sync_duration_milliseconds",
+	"antrea_controller_runtime_info",
+}
+
+var prometheusEnabled bool
+
+// Prometheus server JSON output
+type prometheusServerOutput struct {
+	Status string
+	Data   []map[string]string
+}
+
+func init() {
+	flag.BoolVar(&prometheusEnabled, "prometheus", false, "Enables Prometheus tests")
+}
+
+// skipIfPrometheusDisabled checks if Prometheus testing enabled, skip otherwise
+func skipIfPrometheusDisabled(t *testing.T) {
+	if !prometheusEnabled {
+		t.Skip("Prometheus testing is disabled")
+	}
+}
+
+// getMonitoringAuthToken retrieves monitoring authorization token, required for access to Antrea apiserver/metrics
+// resource
+func getMonitoringAuthToken(t *testing.T, data *TestData) string {
+	secrets, err := data.clientset.CoreV1().Secrets(monitoringNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching monitoring secrets: %v", err)
+	}
+
+	var token string
+	for _, secret := range secrets.Items {
+		if strings.HasPrefix(secret.Name, "prometheus-token") {
+			token = string(secret.Data[v1.ServiceAccountTokenKey])
+		}
+	}
+
+	if len(token) == 0 {
+		t.Fatal("Prometheus ServiceAccount secret not found")
+	}
+
+	return token
+}
+
+// getMetricsFromApiServer retrieves Antrea metrics from Pod apiserver
+func getMetricsFromApiServer(t *testing.T, url string, token string) string {
+	config := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+	tr := &http.Transport{TLSClientConfig: config}
+	client := &http.Client{Transport: tr}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		t.Fatalf("Error creating HTTP request: %v", err)
+	}
+
+	if token != "" {
+		req.Header.Add("Authorization", "Bearer "+token)
+	}
+
+	// Query metrics via HTTPS from Pod
+	resp, err := client.Do(req)
+	if err != nil {
+		t.Fatalf("Error fetching metrics from %s: %v", url, err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("Error retrieving metrics from %s. response: %v", url, err)
+	}
+
+	return string(body)
+}
+
+// testPrometheusMetricsOnPods locate Antrea Pods from the specified component, retrieve and validate that supplied
+// metric exists
+func testPrometheusMetricsOnPods(t *testing.T, data *TestData, component string, metrics []string) {
+	token := getMonitoringAuthToken(t, data)
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=antrea,component=" + component,
+	}
+	pods, err := data.clientset.CoreV1().Pods(antreaNamespace).List(listOptions)
+	if err != nil {
+		t.Fatalf("Error fetching agent Pods: %v", err)
+	}
+
+	var hostIP = ""
+	var hostPort int32 = 0
+	var parser expfmt.TextParser
+
+	// Find Pods' connectivity, check for metrics existence on each of them
```

Comment:
What's the "Pods' connectivity"? You mean the API endpoints?

Reply:
Indeed. I'll change as suggested